sklearn/decomposition/dict_learning.py

```diff
@@ -282,9 +282,9 @@ def sparse_encode(X, dictionary, gram=None, cov=None, algorithm='lasso_lars',
                               check_input=False,
                               verbose=verbose)
         # This ensure that dimensionality of code is always 2,
-        # consistant with the case n_jobs > 1
+        # consistent with the case n_jobs > 1
         if code.ndim == 1:
-            code = code[np.newaxis, :]
+            code = code[:, np.newaxis]
```

Comment:
OK, your logic is that the return dimensions varied whether n_jobs=1 or not, hence this is a clear-cut bug?

I am worried that some users might rely on this behavior. Upgrading scikit-learn might break there code. Do you think it's unlikely? I don't know, I don't personally use this part of scikit-learn.

This is why I was suggesting a more prominent entry in whats_new